src/ImageSharp/Image.FromStream.cs

```diff
@@ -724,7 +724,7 @@ public static Image Load(Configuration configuration, Stream stream, out IImageF
             Guard.NotNull(configuration, nameof(configuration));
             Guard.NotNull(stream, nameof(stream));
 
-            if (!stream.CanRead)
+            if (!stream.CanRead || stream.Length == 0)
```

Comment:
Not sure if you should do this `.Length` check. Don't thinks this will work for non seekable streams.

Reply:
TIL. Will update. https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.length?redirectedfrom=MSDN&view=net-6.0#exceptions

Reply:
Soooo. It turns out those docs are unreliable. [SubReadStream](https://source.dot.net/#System.IO.Compression/System/IO/Compression/ZipCustomStreams.cs,246) is non-seekable but supports the Length property which makes writing a condition impossible.